microsoft-edge/webview2/reference/javascript/hostobjectasyncproxybase.yml

```diff
@@ -0,0 +1,136 @@
+### YamlMime:TSType
+name: HostObjectAsyncProxyBase

```

Comment:
Question for @david-risney - Are these interfaces going to be useful for devs, or are they mostly "internal" implementation details? Is there a way we can make the most common APIs we expect devs to use (probably the HostObjectsOptions?) easier to find or highlight?

Reply:
I made the d.ts file based on patterns I saw in the DOM d.ts file and otherwise got it working so that TS autocomplete worked well. And this was before I saw what our docs tools would produce.

There are a few things that exist to make autocomplete work that could be confusing in our documentation:
1. the EventMap object doesn't actually exist in WebView2 code. That thing is a construct solely for TS autocomplete to be able to map from event name to event handler args so that TS autocomplete can know the difference between `.addEventListener("message", (messageEventArg) => { ... })` and `.addEventListener("whatever", ...);`.
2. The window and chrome objects - we don't really define those ourselves in WebView2. But we need to reference them to show how the webview object maps back to the root object. But having Window and Chrome in our docs is sort of confusing. Would be good to see how other libraries deal with it (if other libraries encroach on the DOM root object)
3. This base thing. The issue is `webview.hostObjects.sync` will get you the sync root proxy, but also all async proxies have a `sync()` method that returns the async proxy as a sync proxy. So I had to pull out this separate base type so that the `webview.hostObjects` isn't ambiguous for `sync()` and `sync`.

All of the above are sort of necessary to have a nice d.ts file when actually writing TS code. And when we put this TS file up online (there's a big d.ts repository that is the defacto standard for finding d.ts files like nuget or npm but for TS) this will be useful.  But when running through the doc tools it results in some weirdness. Fixing the above will probably be a combo of filtering things out of our docs + modifying the original d.ts to make the docs better.

Reply:
@david-risney I've removed the page for Window_2 class. Though WebViewEventMap show up in the method signature [here](https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/javascript/webview?branch=pr-en-us-2315#webview2script-webview-addeventlistener-member(1)). Any idea how we should fix it?